### PR TITLE
MAINT: tidy up Dask implementation

### PR DESCRIPTION
**Pull request by stsievert**
**What does this PR implement?**
Cleans up Dask implementation. It performs the following:

- [x] Makes the client optional in `dist_example.py`. Adding a client will offer additional performance with data/model scattering.
- [x] Removes unused tests.
- [x] Adds Scikit-learn interface
- [x] scale number of workers

Future work:

* Work with Dask arrays.

**References issues/PRs**
This is follow-on work to #2.

### CODE REVIEW

**Comment by stsievert** [comment]
This class tasks Dask workers with computing the gradient. To get a Scikit-learn interface, there are two choices to inherit from: Scikit-learn's BaseEstimator or Skorch's NeuralNet.

It looks like customizing [NeuralNet.train_step_single][1] requires the customization we desire, and the rest of Skorch comes for free. `train_step_single` takes one optimization step: it calculates the loss for given inputs, then populates the gradients. We can override that function to call Dask.

Inheritance from Scikit-learn's BaseEstimator would mean Adadamp would mimic a lot of Skorch's functionality: converting between NumPy ndarrays and PyTorch tensors, getting loss functions and implementing callbacks.

[1]:https://github.com/skorch-dev/skorch/blob/7816c15a3b1986abd433e795b6187973f5ea9987/skorch/net.py#L588-L610

**Comment by stsievert** [comment]
The difference between inheriting from Skorch or Scikit-learn comes down to initializing datasets. Skorch is oriented towards changing the batch size once per epoch. I'll have to fit in that framework. Inheriting from Scikit-learn will be lighter weight, and I won't have to fit into Skorch's framework and depend on their implementation.
